addons/binding/org.openhab.binding.lgtvserial/src/main/java/org/openhab/binding/lgtvserial/internal/LgTvSerialHandlerFactory.java

```diff
@@ -8,42 +8,37 @@
  */
 package org.openhab.binding.lgtvserial.internal;
 
-import static org.openhab.binding.lgtvserial.LgTvSerialBindingConstants.*;
-
-import java.util.Collections;
-import java.util.Set;
-
-import org.openhab.binding.lgtvserial.handler.LgTvSerialHandler;
 import org.eclipse.smarthome.core.thing.Thing;
 import org.eclipse.smarthome.core.thing.ThingTypeUID;
 import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
 import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.openhab.binding.lgtvserial.LgTvSerialBindingConstants;
+import org.openhab.binding.lgtvserial.handler.LgTvSerialHandler;
+import org.openhab.binding.lgtvserial.internal.protocol.serial.SerialCommunicatorFactory;
 
 /**
- * The {@link LgTvSerialHandlerFactory} is responsible for creating things and thing 
+ * The {@link LgTvSerialHandlerFactory} is responsible for creating things and thing
  * handlers.
- * 
+ *
  * @author Marius Bjoernstad - Initial contribution
+ * @author Richard Lavoie - Added communicator to support daisy chained TV
  */
 public class LgTvSerialHandlerFactory extends BaseThingHandlerFactory {
-    
-    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections.singleton(THING_TYPE_LGTV);
-    
+    private final static SerialCommunicatorFactory FACTORY = new SerialCommunicatorFactory();
+
     @Override
     public boolean supportsThingType(ThingTypeUID thingTypeUID) {
-        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+        return thingTypeUID.getBindingId().equals(LgTvSerialBindingConstants.BINDING_ID);
```

Comment:
Looks like this is the only remaining open comment. This kind of implementation may cause issues whenever users make typo's in their Thing definitions in .thing files. 

According to the JavaDoc the original implementation is correct whereas the new one is isn't:

```java
    /**
     * Returns whether the handler is able to create a thing or register a thing handler for the given type.
     *
     * @param thingTypeUID the thing type UID
     * @return true, if the handler supports the thing type, false otherwise
     */
    boolean supportsThingType(ThingTypeUID thingTypeUID);
```

Reply:
My problem with the previous solution is that whenever you add new thing definitions you have to add those ids in the java code as well for it to make sure it handles it. This creates 2 sources of truth and these 2 lists can then become unsynced.

I wanted to prevent what the pioneeravr binding had to do with all the different variants.

Imo, the set in java code is pretty much useless when you already have them defined in an xml file.

Is there a way to fetch the list of ids in the thing definitions file rather than having it defined at 2 places, both in the XML files and in java code ?

Reply:
> Is there a way to fetch the list of ids in the thing definitions file rather than having it defined at 2 places, both in the XML files and in java code ?

The thing types from the XML are available from the `ThingTypeRegistry` but it shouldn't be used in bindings. A binding can have multiple `ThingHandlerFactory`s so it still makes sense to have methods on the factory for querying what types it supports. 

@kaikreuzer do you think it's a good idea for bindings with a single `ThingHandlerFactory` to only check the thing type binding ID?

Reply:
The architecture allows a binding to be split into several bundles, each defining its own thing types and coming with the according handler factory.

In general it is possible that someone implements a custom bundle that defines an additional thing type for this binding that your code does not know about and which it cannot handle.
So I'd agree with @wborn that the original implementation is the correct one, even though it means slightly more work to maintain. To reduce this work, we should rather think about a generator that creates the Java constants from given XML files - which would then be limited to the ones one have created oneself.

Reply:
@wborn @kaikreuzer Why shouldn’t ThingTypeRegistry be used in bindings for them to retrieve the list of things the bundle itself has declared ? Is it just because the thing definition has no idea of which bundle declared it ?

I wonder how many bindings have multiple handler factory or are split in multiple bundles. It would be interesting to get some statistics on that. I suspect most bindings comes as a single bundle and only a few have multipe handler factory.

This could ease and reduce the boilerplate of keeping those 2 lists in sync if we could get the thing definitions back for a given bundle id. 

Note here I’m refering to a bundle id and not the binding id since the binding can be split in multiple bundles and I suspect the binding id will be shared with multiple bundles.

Reply:
I think the APIs bindings are allowed to use are kept intentionally small/limited to reduce the complexity. The more APIs bindings can use the more impact there is when these APIs change. It also increases the likelihood of issues such as cyclic dependencies.